lib/msf/java/rmi/client/jmx/connection/builder.rb

```diff
@@ -0,0 +1,237 @@
+                  operation: -1,
+                  hash: method_hash,
+                  arguments: arguments
+                )
+
+                call
+              end
+
+              # Builds an an array of arguments o build a call to
+              # javax/management/remote/rmi/RMIConnectionImpl_Stub#createMBean()
+              #
+              # @param opts [Hash]
+              # @option opts [String] :name the MBean name
+              # @return [Array]
+              def build_jmx_create_mbean_args(name = '')
+                arguments = [
+                  Rex::Java::Serialization::Model::Utf.new(nil, name),
+                  Rex::Java::Serialization::Model::NullReference.new,
+                  Rex::Java::Serialization::Model::NullReference.new
+                ]
+
+                arguments
+              end
+            end
+
+
+            # Builds an RMI call to javax/management/remote/rmi/RMIConnectionImpl_Stub#invoke()
+            # used to invoke an MBean method
+            #
+            # @param opts [Hash]
+            # @option opts [String] :name the MBean name
+            # @return [Rex::Proto::Rmi::Model::Call]
+            # @see Msf::Java::Rmi::Builder.build_call
+            # @see #build_jmx_invoke_args
+            def build_jmx_invoke(opts = {})
+              object_number = opts[:object_number] || 0
+              uid_number = opts[:uid_number] || 0
+              uid_time = opts[:uid_time] || 0
+              uid_count = opts[:uid_count] || 0
+
+              method_hash = calculate_method_hash('invoke(Ljavax/management/ObjectName;Ljava/lang/String;Ljava/rmi/MarshalledObject;[Ljava/lang/String;Ljavax/security/auth/Subject;)Ljava/lang/Object;')
+
+              arguments = build_jmx_invoke_args(opts)
+
+              call = build_call(
+                object_number: object_number,
+                uid_number: uid_number,
+                uid_time: uid_time,
+                uid_count: uid_count,
+                operation: -1,
+                hash: method_hash,
+                arguments: arguments
+              )
+
+              call
+            end
+
+            # Builds an an array of arguments o build a call to
+            # javax/management/remote/rmi/RMIConnectionImpl_Stub#invoke()
+            #
+            # @param opts [Hash]
+            # @option opts [String] :object the MBean name
+            # @option opts [String] :method the method name
+            # @option opts [Hash] :args the method arguments
+            # @return [Array]
+            def build_jmx_invoke_args(opts = {})
+              object_name = opts[:object] || ''
+              method_name = opts[:method] || ''
+              args = opts[:args] || {}
+
+              builder = Rex::Java::Serialization::Builder.new
+
+              new_object = builder.new_object(
+                name: 'javax.management.ObjectName',
+                serial: 0xf03a71beb6d15cf, # serialVersionUID
```

Comment:
This looks like the same as https://github.com/rapid7/metasploit-framework/pull/5002/files#diff-38fede3a3db37ebfb84bc4f0fe8dc550R53